target/linux/ath79/image/nand.mk

```diff
@@ -211,6 +211,21 @@ define Device/glinet_gl-xe300
 endef
 TARGET_DEVICES += glinet_gl-xe300
 
+define Device/linksys_ea4500-v3
+  SOC := qca9558
+  DEVICE_VENDOR := Linksys
+  DEVICE_MODEL := EA4500
+  DEVICE_VARIANT := v3
+  DEVICE_PACKAGES := kmod-usb2
+  BLOCKSIZE := 128k
+  PAGESIZE := 2048
+  KERNEL_SIZE := 4096k
+  IMAGE_SIZE := 81920k
+  IMAGE/sysupgrade.bin := sysupgrade-tar | append-metadata
+  UBINIZE_OPTS := -E 5
```

Comment:
Maybe you can try whether this factory image can work:
```
  LINKSYS_HWNAME := EA4500V3
  IMAGES += factory.img
  IMAGE/factory.img := append-kernel | pad-to $$$$(KERNEL_SIZE) | \
	append-ubi | check-size | linksys-image type=$$$$(LINKSYS_HWNAME)
```
You need a script to set u-boot envs, please refer to: https://github.com/openwrt/openwrt/blob/master/target/linux/ramips/mt7621/base-files/etc/init.d/bootcount
https://github.com/openwrt/openwrt/commit/f4eef5f2a18400bc7f548d629e1d5825a2b26e5a

Reply:
Done as written below.